### PR TITLE
Refactor/ansible

### PR DESCRIPTION
**Pull request by pythoninthegrass**
Hi @guillaumebriday! Appreciate your work on this!

Made some admittedly opinionated changes — although most of them come from linting and ansible norms — that I thought you might appreciate.

Happy to expand on the following, but these are the high-level modifications:

## docs

- readme: remove non-root `$` prefixes, add ansible.cfg.example, fix typo
- ansible.cfg.example: rename to hosts.ini to match upstream
- hosts.ini.example: add all config for localhost and vars, move become method and user to vars

## refactor

- add become to playbook for non-root sudoers
- FQCNs (builtin vs. community)
- fix docker gpg key (keyrings directory, implicit octal notation)
- tags (selectively run roles)
- add ansible.cfg.example w/facts directory placeholder
- add .ansible-lint
- update .gitignore

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,35 +12,37 @@ It will automatically update your packages and configure these packages to secur
...
+cp ansible.cfg.example ansible.cfg
 ```
 
+Update the `<host1>` in `hosts.ini` with your server's IP address (you can have multiple servers):
+
 Run the playbook:
+
 ```bash
-$ ANSIBLE_HOST_KEY_CHECKING=False ansible-playbook -i hosts.ini playbook.yml
+ANSIBLE_HOST_KEY_CHECKING=False ansible-playbook -i hosts.ini playbook.yml
```
Comment (pythoninthegrass) [inline README.md:40]: tbf the inline env var and hosts arg aren't necessary with `ansible.cfg` left for posterity / if end users don't use the `ansible.cfg.example`

File: roles/docker/tasks/main.yml [hunk 2]
```diff
@@ -1,32 +1,43 @@
...
+    mode: "0755"
+  tags: docker
+
+- name: Add Docker GPG apt key
+  ansible.builtin.get_url:
     url: "https://download.docker.com/linux/ubuntu/gpg"
-    dest: /etc/apt/trusted.gpg.d/docker.asc
-    mode: 0644
-    force: false
+    dest: /etc/apt/keyrings/docker.asc
```
Comment (pythoninthegrass) [inline roles/docker/tasks/main.yml:21]: This isn't arbitrary. Got some errors locally on 24.04.1 and the [official docs](https://docs.docker.com/engine/install/ubuntu/#install-using-the-repository) store the gpg key in `/etc/apt/keyrings`:

```bash
# Add Docker's official GPG key:
sudo apt-get update
sudo apt-get install ca-certificates curl
sudo install -m 0755 -d /etc/apt/keyrings
sudo curl -fsSL https://download.docker.com/linux/ubuntu/gpg -o /etc/apt/keyrings/docker.asc
sudo chmod a+r /etc/apt/keyrings/docker.asc
```

File: roles/packages/tasks/main.yml [hunk 3]
```diff
@@ -20,9 +21,10 @@
       - unattended-upgrades
       - vim
     state: latest
-    update_cache: yes
-    autoremove: yes
-    autoclean: yes
+    update_cache: true
```
Comment (pythoninthegrass) [inline roles/packages/tasks/main.yml:24]: Booleans are truthy

File: .gitignore [hunk 4]
```diff
@@ -1 +1,36 @@
-hosts.ini
+# ETC
+ansible.cfg
+hosts*
```
Comment (guillaumebriday) [inline .gitignore:3]: I guess we can only keep these two, the rest should be in a global gitignore and are not specific to this project 👍

File: .ansible-lint [hunk 5]
```diff
@@ -0,0 +1,17 @@
+---
```
Comment (guillaumebriday) [inline .ansible-lint:1]: Great, but maybe in an other PR?

File: hosts.ini.example [hunk 6]
```diff
@@ -1,2 +1,9 @@
+[all]
+localhost               ansible_connection=local
```
Comment (guillaumebriday) [inline hosts.ini.example:2]: is this group needed?

File: roles/docker/tasks/main.yml [hunk 7]
```diff
@@ -1,32 +1,43 @@
...
-  package:
+  ansible.builtin.package:
     name:
       - docker-ce
       - docker-ce-cli
       - containerd.io
     state: present
   notify:
     - restart docker
+  tags: docker
```
Comment (guillaumebriday) [inline roles/docker/tasks/main.yml:43]: I think tags are a great idea, but maybe kinda overkill for this project, let's keep it simple

File: roles/packages/tasks/main.yml (hunk 3 above)
Comment (guillaumebriday) [reply roles/packages/tasks/main.yml:24]: Hum both works, must of examples in the documentation use `yes` instead of `true`

**Comment by guillaumebriday** [comment]
Please rebase the branch if you still want to submit changes in this PR 🙏 

Closing in the meantime, feel free to reopen

**Pull request closed without merging**